core/service/src/client/tests/threshold/custodian_backup_tests.rs

```diff
@@ -527,17 +529,20 @@ async fn decrypt_after_recovery_negative(amount_custodians: usize, threshold: u3
 #[cfg(feature = "insecure")]
 async fn run_custodian_recovery_init(
     kms_clients: &HashMap<u32, CoreServiceEndpointClient<Channel>>,
-) -> Vec<RecoveryRequest> {
+) -> Vec<(u32, RecoveryRequest)> {
```

Comment:
Why the use of role index and not verification key here? I think this leads back to a previous comment about storing `kms_clients` in a hash map keyed with the verification key instead of role index

Reply:
answered here https://github.com/zama-ai/kms/pull/296#discussion_r2571954813

I had to add this u32 to link the request to the key used in the core-client map